grtc/DESIGN.md

```diff
@@ -0,0 +1,193 @@
+# Design
```

Comment:
Also @gziolo maybe some of these notes can be added as comments in the code, which is always better.

Reply:
I updated the docs, they may require further iterations. Let's wait until we stabilize API.